File: service-proxy-examples/service-consumer/src/main/js/io/vertx/examples/service/consumer/consumer_verticle.js
var ProcessorService = require("vertx-processor-sample-js/processor_service");
var service = ProcessorService.createProxy(vertx, "vertx.processor");

var document = {
  "name" : "vertx"
};

service.process(document, function (r, r_err) {
  if (r_err == null) {
    console.log(JSON.stringify(r));
  } else {
    if (r_err.getClass().getSimpleName() == 'ServiceException') {
      var exc = r_err;
      if (exc.failureCode() === BAD_NAME_ERROR) {
        console.log("Failed to process the document: The name in the document is bad. The name provided is: " + exc.getDebugInfo().name);
      } else if (exc.failureCode() === NO_NAME_ERROR) {
        console.log("Failed to process the document: No name was found");
      }
    } else {
      console.log("Unexpected error: " + r_err);

    }
  }
});

File: service-proxy-examples/service-consumer/src/main/rb/io/vertx/examples/service/consumer/consumer_verticle.rb
require 'json'
require 'vertx-processor-sample/processor_service'
service = VertxProcessorSample::ProcessorService.create_proxy($vertx, "vertx.processor")

document = {
  'name' => "vertx"
}

service.process(document) { |r_err,r|
  if (r_err == nil)
    puts JSON.generate(r)
  else
    Java::IoVertxExamplesServiceConsumer::Failures.deal_with_failure(r_err)
  end
}
